opengrok-tools/src/main/python/opengrok_tools/scm/repository.py

```diff
@@ -75,6 +82,105 @@ def incoming(self):
         """
         Check if there are any incoming changes.
 
+        Return True if so, False otherwise.
+        """
+        if self.is_command_overridden(self.configured_commands, 'incoming'):
+            return self._run_custom_incoming_command(
+                self.listify(self.configured_commands['incoming'])
+            )
+        return self.incoming_check()
+
+    def incoming_check(self):
+        """
+        Check if there are any incoming changes.
+
         Return True if so, False otherwise.
         """
         return True
+
+    def _run_custom_sync_command(self, command):
+        """
+        Execute the custom sync command.
+
+        :param command: the command
+        :return: 0 on success execution, 1 otherwise
+        """
+        status, output = self._run_command(command)
+        log_handler = self.logger.info if status == 0 else self.logger.warning
+        log_handler("output of '{}':".format(command))
+        log_handler(output)
+        return status
+
+    def _run_custom_incoming_command(self, command):
+        """
+        Execute the custom incoming command.
+
+        :param command: the command
+        :return: true when there are changes, false otherwise
+        """
+        status, output = self._run_command(command)
+        if status != 0:
+            self.logger.error("output of '{}':".format(command))
+            self.logger.error(output)
+            raise RepositoryException(
+                'failed to check for incoming in repository {}'.format(self)
+            )
+        return len(output.strip()) > 0
+
+    def _run_command(self, command):
+        """
+        Execute the command.
+
+        :param command: the command
+        :return: tuple of (status, output)
+                    - status: 0 on success execution, non-zero otherwise
+                    - output: command output as string
+        """
+        cmd = self.getCommand(command, work_dir=self.path,
+                              env_vars=self.env, logger=self.logger)
+        cmd.execute()
+        if cmd.getretcode() != 0 or cmd.getstate() != Command.FINISHED:
+            cmd.log_error("failed to perform command")
+            status = cmd.getretcode()
+            if status == 0 and cmd.getstate() != Command.FINISHED:
+                status = 1
+            return status, '\n'.join(filter(None, [
+                cmd.getoutputstr(),
+                cmd.geterroutputstr()
+            ]))
+        return 0, cmd.getoutputstr()
+
+    @staticmethod
+    def _repository_command(configured_commands, default=lambda: None):
+        """
+        Get the repository command, or use default supplier.
+
+        :param configured_commands: commands section from configuration
```

Comment:
this should probably say that it looks up the 'command' key

Reply:
when saying that I should probably mention that it does lookup only when configuration is a dict and that is exactly what is written in the code...so I'm not a big fan rephrasing the code to the comment

Reply:
When I add that, I should perhaps mention that it only does the lookup when the configured_commands is a dictionary and that all together....is exactly what is in the code. So I'm not a big fan adding that comment.